Resources/public/js/views/dashboard/ez-dashboardblockallcontentview.js

```diff
@@ -0,0 +1,167 @@
+            this.events = Y.merge(this.events, EVENTS);
+
+            this._set('identifier', BLOCK_IDENTIFIER);
+            this.get('container')
+                .addClass(this._generateViewClassName(Y.eZ.DashboardBlockBaseView.NAME))
+                .addClass(CLASS_LOADING);
+
+            this.after('itemsChange', this.render);
+            this.after('itemsChange', this._uiEndLoading);
+        },
+
+        /**
+         * Renders the dashboard block view
+         *
+         * @method render
+         * @return {eZ.DashboardBlockAllContentView} the view itself
+         */
+        render: function () {
+            var items = this.get('items').map(function (item) {
+                    return {
+                        content: item.content.toJSON(),
+                        contentType: item.contentType.toJSON(),
+                        location: item.location.toJSON()
+                    };
+                });
+
+            this.get('container').setHTML(this.template({items: items}));
+
+            return this;
+        },
+
+        /**
+         * Shows the row options
+         *
+         * @method _uiShowOptions
+         * @protected
+         * @param event {Object} event facade
+         */
+        _uiShowRowOptions: function (event) {
+            this._uiHideRowOptions();
+
+            event.currentTarget.addClass(CLASS_ROW_SELECTED);
+
+            this._clickOutsideHandler = this.get('container').one(SELECTOR_CONTENT).on('clickoutside', Y.bind(this._uiHideRowOptions, this));
+        },
+
+        /**
+         * Clears the click outside handler
+         *
+         * @method _clearClickOutsideHandler
+         * @protected
+         */
+        _clearClickOutsideHandler: function () {
+            if (this._clickOutsideHandler) {
+                this._clickOutsideHandler.detach();
+            }
+        },
+
+        /**
+         * Hides the row options
+         *
+         * @method _uiHideOptions
+         * @protected
+         */
+        _uiHideRowOptions: function () {
+            this.get('container').all(SELECTOR_ROW).removeClass(CLASS_ROW_SELECTED);
+
+            this._clearClickOutsideHandler();
+        },
+
+        /**
+         * Removes the loading state of the UI
+         *
+         * @method _uiEndLoading
+         * @protected
+         */
+        _uiEndLoading: function () {
+            this.get('container').removeClass(CLASS_LOADING);
+        },
+
+        /**
+         * Makes request for data
+         *
+         * @method _fireSearchEvent
+         * @protected
+         * @param event {Object} event facade
+         */
+        _fireSearchEvent: function () {
+            var rootLocation = this.get('rootLocation');
+
+            this.fire('locationSearch', {
+                viewName: 'all-content-' + rootLocation.get('locationId'),
+                resultAttribute: 'items',
+                loadContentType: true,
+                loadContent: true,
```

Comment:
too bad the `currentVersionNo` is not available in the ContentInfo (see https://jira.ez.no/browse/EZP-25842), that would avoid having to load the Content. Can you add a TODO to remove the `loadContent` flag with a link to that JIRA issue, that way if manage to that in a very near feature, we can remove that.